server/helpers/multer/posts.js

```diff
@@ -0,0 +1,32 @@
+const multer = require('multer');
+
+//defining multer's diskStorage:
+const storage = multer.diskStorage({
+	destination: function (req, file, callback) {
+		//null if there's an error and path where we store the file:
+		callback(null, './server/uploads/posts');
+	},
+	filename: function (req, file, callback) {
+		//the date that we uploaded the file(replacing the ":" with "-" because of windows problem) with the originalname of the file:
+		//toISOString document: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
+		callback(null, new Date().toISOString().replace(/:/g, '-') + file.originalname);
+	}
+});
+
+//rejecting a file if it's not the type of file we want:
+const fileFilter = (req, file, cb) => {
+	if(file.mimetype === 'image/png' || file.mimetype === 'image/jpg' || file.mimetype === 'image/jpeg' || file.mimetype === 'video/mp4') {
+        //second parameter is a boolean which will either save the file or reject it(true or false):
+		cb(null, true);
+	} else {
+		cb(new Error('Type file is not supported!'), false);
+	}
+}
+
+//a variable(upload) with everything multer related we defined above:
+const upload = multer ({
+	storage: storage,
+	fileFilter: fileFilter
+})
```

Comment:
I think this entire thing could be way shorter. example: 

```
const post = multer({
	fileFilter(req, file, cb) {
		if (!file.originalname.match(/\.(jpg|jpeg|png|bmp|mov|mp4)$/g)) {
			return cb(new Error('Please upload an image/video'));
		}

		cb(undefined, true);
	}
});
```
and then the routes can handle the errors: 

`
app.post(
		'/api/posts',
		verifyUser,
		post.single('post'),
		async (req, res) => {
			if (!req.body) {
				return res
					.status(400)
					.json({ message: `request is invalid` })
					.end();
			}

			const buffer = req.file.buffer

			const post = {
				...req.body,
				user  : req.user.sub,
				media : buffer
			};
			try {
				const newPost = await createPost(post);
				res.status(200).json(newPost).end();
			} catch (e) {
				res
					.status(500)
					.json({
						message : `internal error while trying to create post`
					})
					.end();
			}
		},
		**(error, req, res, next) => {
			res.status(400).json({ error: error.message }).end();
		}
	);**
`

If we save the binary data on the instances of the posts, we don't need to set destination and path just a fileFilter

Reply:
Also for profile pics, you should implement [sharp](https://www.npmjs.com/package/sharp) so we can set a fixed width and heigh of like 180px or something and also a fixed format of .jpg so when we set up a GET route for the profile pic we can just send a fixed JPG or JPEG without having to accomodate for every single format in the world like you tried to do in line 18 (server/helpers/multer/posts.js)